templates/terraform/examples/region_network_endpoint_group_basic.tf.erb

```diff
@@ -0,0 +1,121 @@
+// Cloud Run Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "run-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  cloud_run {
+    service = google_cloud_run_service.<%= ctx[:primary_resource_id] %>.name
+  }
+}
+
+resource "google_cloud_run_service" "<%= ctx[:primary_resource_id] %>" {
+  name     = "hello-world"
+  location = "us-central1"
+
+  template {
+    spec {
+      containers {
+        image = "gcr.io/cloudrun/hello"
+      }
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+}
+
+
+// Cloud Functions Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "gcf-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  cloud_function {
+    function = google_cloudfunctions_function.<%= ctx[:primary_resource_id] %>.name
+  }
+}
+
+resource "google_cloudfunctions_function" "<%= ctx[:primary_resource_id] %>" {
+  name        = "function-test"
+  description = "My function"
+  runtime     = "nodejs10"
+
+  available_memory_mb   = 128
+  source_archive_bucket = google_storage_bucket.bucket.name
+  source_archive_object = google_storage_bucket_object.archive.name
+  trigger_http          = true
+  entry_point           = "helloGET"
+}
+
+// App Engine Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "gae-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  app_engine {
+    service = google_app_engine_flexible_app_version.<%= ctx[:primary_resource_id] %>.service
+    version = google_app_engine_flexible_app_version.<%= ctx[:primary_resource_id] %>.version_id
+  }
+}
+
+resource "google_app_engine_flexible_app_version" "<%= ctx[:primary_resource_id] %>" {
+  version_id = "v1"
+  service    = "default"
+  runtime    = "nodejs"
+
+  entrypoint {
+    shell = "node ./app.js"
+  }
+
+  deployment {
+    zip {
+      source_url = "https://storage.googleapis.com/${google_storage_bucket.bucket.name}/${google_storage_bucket_object.object.name}"
+    }
+  }
+
+  liveness_check {
+    path = "/"
+  }
+
+  readiness_check {
+    path = "/"
+  }
+
+  env_variables = {
+    port = "8080"
+  }
+
+  handlers {
+    url_regex        = ".*\\/my-path\\/*"
+    security_level   = "SECURE_ALWAYS"
+    login            = "LOGIN_REQUIRED"
+    auth_fail_action = "AUTH_FAIL_ACTION_REDIRECT"
+
+    static_files {
+      path = "my-other-path"
+      upload_path_regex = ".*\\/my-path\\/*"
+    }
+  }
+
+  automatic_scaling {
+    cool_down_period = "120s"
+    cpu_utilization {
+      target_utilization = 0.5
+    }
+  }
+
+  noop_on_destroy = true
+}
+
+resource "google_storage_bucket" "bucket" {
+  project = google_project.my_project.project_id

```

Comment:
This isn't a reference to a resource that exists in this file. These examples become integration tests for us, so they need to work. Take a look at https://github.com/GoogleCloudPlatform/magic-modules#testing-your-changes for more details on how to test your changes.